art/attacks/evasion/fast_gradient.py

```diff
@@ -334,8 +334,11 @@ def generate(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> n
 
     def _check_params(self) -> None:
 
-        if self.norm not in [1, 2, np.inf, "inf"]:
-            raise ValueError('Norm order must be either 1, 2, `np.inf` or "inf".')
+        if not (
+            self.norm == "inf"
+            or self.norm >= 1
```

Comment:
The type checking with `mypy` is complaining here because it does not recognise that `self.norm` can only be `int` on the right side of the `or`. Maybe it could work by translating this statement into an `and` of two `not` instead of using `or` inside `not`?

Full mypy report: https://github.com/Trusted-AI/adversarial-robustness-toolbox/actions/runs/8565558666/job/23507533560?pr=2382

Reply:
Since the problem occurs many times, I prefer creating a float variable
```python
norm: float = np.inf if self.norm == "inf" else float(norm)
```
when necessary.